source/validate.cpp

```diff
@@ -279,6 +475,19 @@ spv_result_t spvValidate(const spv_const_context context,
     return SPV_ERROR_INVALID_BINARY;
   }
 
+  ValidationState_t vstate(pDiagnostic);
+  auto err = spvBinaryParse(context, &vstate, binary->code, binary->wordCount,
+                            setHeader, pushInstructions, pDiagnostic);
+
+  if (err) {
```

Comment:
And here say "Now validate things which require the entire module to have been seen.  Use the data we collected on the single parsing pass through the module."
But only if that's what the design is going to be.  :-)